microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -90,6 +91,47 @@ def verify_dpdk_build_failsafe(
     ) -> None:
         self._verify_dpdk_build(node, log, variables, "failsafe")
 
+    @TestCaseMetadata(
+        description="""
+           Install and run OVS+DPDK functional tests
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            min_nic_count=2,
+            network_interface=Sriov(),
+        ),
+    )
+    def verify_dpdk_ovs(
+        self, node: Node, log: Logger, variables: Dict[str, Any]
+    ) -> None:
+        # initialize DPDK first, OVS requires it built before configuring.
+        test_kit = initialize_node_resources(node, log, variables, "failsafe")
+
+        # checkout OpenVirtualSwitch
+        ovs = node.tools[DpdkOvs]
+
+        # provide ovs build with DPDK tool info and build
+        ovs.build_with_dpdk(test_kit.testpmd)
+
+        # enable hugepages needed for dpdk EAL
+        _init_hugepages(node)
```

Comment:
BTW, I implemented it within xdp too. Can you merge them? It needs to be removed after test case is done, or it may cause conflicts among test cases. It needs to consider which one (xdp, dpdk) depends on another one, to decide the place of the code. If it takes longer time, you can try it in next PR.

Reply:
I think this is a good idea but doing it correctly will be a bigger commit. I think we would want to implement a meminfo tool to get the initial memory state, then add hugepages functions as a second tool.

I'll create a work item for it, I think it's a bigger change than I would want to include here.

Reply:
the hugepages is not a command likely. We can add a virtual tool, but it may be confused. Because it just wraps our test data, not some common tool. So we can put it in the performance or some test suite folder like common, so it can be reused.